pydantic/types.py

```diff
@@ -127,6 +145,45 @@ def conbytes(*, strip_whitespace: bool = False, min_length: int = None, max_leng
     return type('ConstrainedBytesValue', (ConstrainedBytes,), namespace)
 
 
+T = TypeVar("T")
+
+
+class ConstrainedList(List[T]):
+    __origin__ = list  # Needed for pydantic to detect that this is a list
+
+    min_items: Optional[int] = None
+    max_items: Optional[int] = None
+    item_type: Type[T]
+
+    @classmethod
+    def __get_validators__(cls) -> 'CallableGenerator':
+        yield cls.list_length_validator
+
+    @classmethod
+    def list_length_validator(cls, v: 'List[T]', field: 'Field', config: 'BaseConfig') -> 'List[T]':
+        v_len = len(v)
+
+        min_items = cls.min_items
```

Comment:
no need to define this as a separate variable, and below.

Reply:
Done